corehq/apps/hqwebapp/static/hqwebapp/js/ui_elements/ui-element-key-val-mapping.js

```diff
@@ -1,5 +1,32 @@
-/* globals hqDefine, hqImport, $, _, django, ko */
+/**
+    Key-value mapping UI element.
 
+    This element generates a button that pops up a modal containing pairs of keys and values.
+    Expects to be used in app manager, in the context of an application module.
+    Used primarily in case list/detail display properties. Keys may be strings
+    (as in ID Mapping format) or calcuations (as in Conditional ID Mapping format).
+    Values may be strings (as in both ID Mapping formats) or references to images
+    (as in Icon format). Supports translations.
+
+    Options:
+        Required:
+            items: Array of current values. Each item should be the javascript equivalent
+                of a MappingItem.
+            lang: Current language.
+            langs: All languages.
+            property_name: Observable representing the name of the overall mapping.
+                May be either a string or one of the elements defined in hqwebapp/js/ui-element.
+                Used primarily to add title to modal. If values_are_icons, this should be a slug.
+        Optional:
+            keys_are_conditions: True iff KEYS are calculations rather than strings.
+            values_are_icons: True iff values are image references ("jr://file...")
```

Comment:
What if one language has an image path, but others are plain text?

Also what if they should be audio/video instead of icons?

Reply:
Er...neither of those things are supported? Despite this file being in hqwebapp, the image-specific parts of it are pretty closely coupled with the icon format in case list/detail display properties (for example the expectation that you provide a `module_id`).